sdks/go/pkg/beam/runners/prism/internal/jobservices/management.go

```diff
@@ -243,21 +243,21 @@ func (s *Server) Run(ctx context.Context, req *jobpb.RunJobRequest) (*jobpb.RunJ
 // Otherwise, returns nil if Job does not exist or the Job's existing state as part of the CancelJobResponse.
 func (s *Server) Cancel(_ context.Context, req *jobpb.CancelJobRequest) (*jobpb.CancelJobResponse, error) {
 	s.mu.Lock()
+	defer s.mu.Unlock()
 	job, ok := s.jobs[req.GetJobId()]
-	s.mu.Unlock()
 	if !ok {
 		return nil, nil
 	}
 	state := job.state.Load().(jobpb.JobState_Enum)
 	switch state {
-	case jobpb.JobState_CANCELLED, jobpb.JobState_DONE, jobpb.JobState_DRAINED, jobpb.JobState_UPDATED, jobpb.JobState_FAILED:
+	case jobpb.JobState_CANCELLED, jobpb.JobState_DONE, jobpb.JobState_DRAINED, jobpb.JobState_UPDATED, jobpb.JobState_FAILED, jobpb.JobState_STOPPED:
 		// Already at terminal state.
 		return &jobpb.CancelJobResponse{
 			State: state,
 		}, nil
 	}
 	job.SendMsg("canceling " + job.String())
-	job.Canceling()
+	job.Canceled()
```

Comment:
Nit: A job can move straight to canceled, but only if all resources are already cleaned up (SDKs, docker environments, etc). Is that true at this point?

Reply:
The underlying cause of me doing this was poorly written tests from misunderstanding the various lifecycles and when they should occur.